stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -262,30 +265,33 @@ class IMetadataProvider(Protocol):
     def metadata_listdir(self, name: str) -> list[str]: ...
     def get_metadata(self, name: str) -> str: ...
     def get_metadata_lines(self, name: str) -> Generator[str, None, None]: ...
-    def run_script(self, script_name: str, namespace: dict[str, Any]) -> None: ...
+    def run_script(self, script_name: str, namespace: Mapping[str, object]) -> None: ...
 
 class ResolutionError(Exception): ...
 
 class DistributionNotFound(ResolutionError):
+    def __init__(self, req: Requirement, requirers: set[str] | None, /, *args: object) -> None: ...
```

Comment:
`ResolutionError` subclasses still work with any number of arguments wich will show up in the `__repr__` string